x-pack/plugins/streams/server/lib/streams/stream_crud.ts

```diff
@@ -424,6 +502,138 @@ export async function syncStream({
+  const unmanagedAssets = await getUnmanagedElasticsearchAssets({
+    dataStream,
+    scopedClusterClient,
+  });
+  const executionPlan: ExecutionPlanStep[] = [];
+  const streamManagedPipelineName = getClassicPipelineName(definition.id);
+  const pipelineName = unmanagedAssets.find((asset) => asset.type === 'ingest_pipeline')?.id;
+  if (!pipelineName) {
+    throw new Error('Unmanaged stream needs a default ingest pipeline');
+  }
+  if (pipelineName === streamManagedPipelineName) {
+    throw new Error('Unmanaged stream cannot have the @stream pipeline as the default pipeline');
+  }
+  await findStreamManagedPipelineReferenceAddOrCheckNestedCustomPipeline(
+    scopedClusterClient,
+    pipelineName,
+    definition,
+    executionPlan
+  );
+
+  if (definition.processing.length) {
+    // if the stream has processing, we need to create or update the stream managed pipeline
+    executionPlan.push({
+      method: 'PUT',
+      path: `/_ingest/pipeline/${streamManagedPipelineName}`,
+      body: generateClassicIngestPipelineBody(definition),
+    });
+  } else {
+    const pipelineExists = Boolean(
+      await tryGettingPipeline({ scopedClusterClient, id: streamManagedPipelineName })
+    );
+    // no processing, just delete the pipeline if it exists. The reference to the pipeline won't break anything
+    if (pipelineExists) {
+      executionPlan.push({
+        method: 'DELETE',
+        path: `/_ingest/pipeline/${streamManagedPipelineName}`,
+      });
+    }
+  }
+
+  for (const step of executionPlan) {
+    await scopedClusterClient.asCurrentUser.transport.request({
+      method: step.method,
+      path: step.path,
+      body: step.body,
+    });
+  }
+}
+
+async function findStreamManagedPipelineReferenceAddOrCheckNestedCustomPipeline(
+  scopedClusterClient: IScopedClusterClient,
+  pipelineName: string,
+  definition: StreamDefinition,
+  executionPlan: ExecutionPlanStep[]
+) {
+  const streamManagedPipelineName = getClassicPipelineName(definition.id);
+  const pipeline = (await tryGettingPipeline({ scopedClusterClient, id: pipelineName })) || {
+    processors: [],
+  };
+  const streamProcessor = pipeline.processors?.find(
+    (processor) => processor.pipeline && processor.pipeline.name === streamManagedPipelineName
+  );
+  const customProcessor = pipeline.processors?.findLast(
+    (processor) => processor.pipeline && processor.pipeline.name.endsWith('@custom')
+  );
+  if (streamProcessor) {
+    return;
+  }
+  if (customProcessor) {
+    // go one level deeper, find the latest @custom leaf pipeline
+    await findStreamManagedPipelineReferenceAddOrCheckNestedCustomPipeline(
+      scopedClusterClient,
+      customProcessor.pipeline!.name,
+      definition,
+      executionPlan
+    );
+    return;
+  }
+  const callStreamManagedPipelineProcessor: IngestProcessorContainer = {
+    pipeline: {
+      name: streamManagedPipelineName,
+      if: `ctx._index == '${definition.id}'`,
+      ignore_missing_pipeline: true,
```

Comment:
Can we add `description` to this processor that says something like "This is processor is being managed by Streams. If you change OR remove this processor step you will be introducing a breaking change to your index configurations."